nuxeo-services/nuxeo-platform-types-core/src/main/java/org/nuxeo/ecm/platform/types/TypeRegistry.java

```diff
@@ -236,4 +245,39 @@ public Type getType(String id) {
         return types.get(id);
     }
 
+    /**
+     * @since 8.4
+     */
+    protected void updateCoreContribution(String id, Type contrib) {
+        SchemaManagerImpl schemaManager = (SchemaManagerImpl) Framework.getService(SchemaManager.class);
+
+        // if there's already a core contribution, unregiser it and register a new one
+        if (dtds.containsKey(id)) {
+            schemaManager.unregisterDocumentType(dtds.get(id));
```

Comment:
Won't you have an issue there if you unregister a DocumentTypeDescriptor for a given type which has some facets, schemas etc as you probably won't contribute a new "UI type contribution" with schemas, facets, etc?

I would like to see a test here that proves I'm wrong :)